libs/platform/ebpf_serialize.c

```diff
@@ -179,3 +214,384 @@ ebpf_deserialize_map_information_array(
+    if (result != EBPF_SUCCESS)
+        goto Exit;
+    local_helper_prototype_array = (ebpf_helper_function_prototype_t*)ebpf_allocate(helper_prototype_array_size);
+    if (local_helper_prototype_array == NULL) {
+        result = EBPF_NO_MEMORY;
+        goto Exit;
+    }
+    local_program_info->helper_prototype = local_helper_prototype_array;
+
+    for (uint16_t helper_function_index = 0; helper_function_index < helper_function_count; helper_function_index++) {
+        ebpf_serialized_helper_function_prototype_t* serialized_helper_prototype =
+            (ebpf_serialized_helper_function_prototype_t*)current;
+        ebpf_helper_function_prototype_t* helper_prototype = &local_helper_prototype_array[helper_function_index];
+        char* local_helper_function_name;
+
+        // Check if sufficient input buffer left for ebpf_serialized_helper_function_prototype_t.
+        if (buffer_left < sizeof(ebpf_serialized_helper_function_prototype_t)) {
+            result = EBPF_INVALID_ARGUMENT;
+            goto Exit;
+        }
+
+        // Serialize helper prototype.
+        helper_prototype->helper_id = serialized_helper_prototype->helper_id;
+        helper_prototype->return_type = serialized_helper_prototype->return_type;
+        for (int i = 0; i < 5; i++)
+            helper_prototype->arguments[i] = serialized_helper_prototype->arguments[i];
+
+        // Adjust remaining buffer length.
+        result = ebpf_safe_size_t_subtract(
+            buffer_left, EBPF_OFFSET_OF(ebpf_serialized_helper_function_prototype_t, name), &buffer_left);
+        if (result != EBPF_SUCCESS)
+            goto Exit;
+
+        // Check if enough buffer left for helper function name.
+        if (buffer_left < serialized_helper_prototype->name_length) {
+            result = EBPF_INVALID_ARGUMENT;
+            goto Exit;
+        }
+
+        // Allocate buffer and serialize helper function name.
+        local_helper_function_name = (char*)ebpf_allocate(serialized_helper_prototype->name_length + 1);
+        if (local_helper_function_name == NULL) {
+            result = EBPF_NO_MEMORY;
+            goto Exit;
+        }
+        memcpy(local_helper_function_name, serialized_helper_prototype->name, serialized_helper_prototype->name_length);
+        helper_prototype->name = local_helper_function_name;
+
+        // Adjust remaining buffer length.
+        result = ebpf_safe_size_t_subtract(buffer_left, serialized_helper_prototype->name_length, &buffer_left);
+        if (result != EBPF_SUCCESS)
+            goto Exit;
+
+        // Advance the current pointer to the end of the helper function prototype.
+        current += serialized_helper_prototype->size;
+    }
+
+Exit:
+    if (result == EBPF_SUCCESS) {
+        *program_info = local_program_info;
+        local_program_info = NULL;
+    }
+
+    if (local_program_info != NULL) {
+        ebpf_program_information_free(local_program_info);
+    }
```

Comment:
make semantics match free() for consistency, i.e., move the NULL check inside the function